src/sincere_singularities/data/db.py

```diff
@@ -0,0 +1,225 @@
+        self.connected = False
+        try:
+            self.client = MongoClient(DB_URI)
+            self.connected = True
+            self.client.server_info()
+            self.db = self.client[DB_NAME]
+            self.restaurants = self.db.restaurants
+        except errors.ConnectionFailure as err:
+            self.connected = False
+            print(f"Error: {err}")
+            raise err from None
+
+    def is_connected(self) -> bool:
+        """Check if connected
+
+        Returns:
+            bool: True if connected, False otherwise
+        """
+        return self.connected
+
+    def add_element(self, collection: str, data: dict[str, Any]) -> None:
+        """Add element
+
+        Args:
+            collection (str): Collection name
+            data (dict[str, Any]): Data to add
+
+        Returns:
+            None
+        """
+        if not self.connected:
+            raise ConnectError("Not connected to the database")
+
+        name = data["name"]
+        # Check if the restaurant already exists
+        if not self.db[collection].find_one({"name": name}):
+            current_time = datetime.now(utc_timezone)
+            data["created_at"] = current_time
+            data["updated_at"] = current_time
+            self.db[collection].insert_one(data)
+
+    def add_many(self, collection: str, datas: Iterable[Any]) -> None:
+        """Add many elements
+
+        Args:
+            collection (str): Collection name
+            datas (Iterable[Any]): list of Datas to add
+
+        Returns:
+            None
+        """
+        for data in datas:
+            self.add_element(collection, data)
+
+    def delete_all(self, collection: str) -> None:
+        """Delete all elements
+
+        Args:
+            collection (str): Collection name
+
+        Returns:
+            None
+        """
+        if not self.connected:
+            raise ConnectError("Not connected to the database")
+        self.db[collection].delete_many({})
+
+    def delete_many(self, collection: str, datas: Iterable[Any]) -> None:
+        """Delete many elements
+
+        Args:
+            collection (str): Collection name
+            datas (Iterable[Any]): list of datas to delete
+
+        Returns:
+            None
+        """
+        if not self.connected:
+            raise ConnectError("Not connected to the database")
+
+        for data in datas:
+            if self.show_one(collection, data):
+                self.db[collection].delete_one(data)
+
+    def delete_one(self, collection: str, data: dict[str, Any]) -> None:
+        """Delete one element
+
+        Args:
+            collection (str): Collection name
+            data (dict[str, Any]): Data to delete
+
+        Returns:
+            None
+        """
+        if not self.connected:
+            raise ConnectError("Not connected to the database")
+        if self.show_one(collection, data):
+            self.db[collection].delete_one(data)
```

Comment:
Maybe raise an exception if it didn't exist in the first place? (also in `delete_many`)